app.py

```diff
@@ -18,6 +18,160 @@
 """This is the main script of the template project."""
 
 from template.version import __version__
+import logging
+import sys
+from datetime import date
+from datetime import datetime
+from typing import Dict
+from typing import Any
+from typing import Optional
+from urllib.parse import urlparse
 
-if __name__ == "__main__":
-    print(f"A template project with Thoth integration, v{__version__}.")
+import requests
+import click
+import yaml
+from thoth.common import init_logging
+from thoth.storages import SolverResultsStore
+
+init_logging()
+_LOGGER = logging.getLogger("thoth.prescriptions.gh_release_notes")
+_LOGGER.setLevel(logging.INFO)
+_DATE_FORMAT = "%Y-%m-%d"
+
+__component_version__ = f"{__version__}"
+
+
+def get_source_repos(*, start_date: Optional[date], end_date: Optional[date]) -> Dict[str, Any]:
+    """Get source URLs of github repos."""
+    store = SolverResultsStore()
+    store.connect()
+
+    git_source_repos = {}
+    for document_id, doc in store.iterate_results(start_date=start_date, end_date=end_date, include_end_date=True):
+        if not doc["result"]["tree"]:
+            continue
+        _LOGGER.debug("Processing solver document %r", document_id)
+
+        metadata = doc["result"]["tree"][0]["importlib_metadata"]["metadata"]
+        name = metadata.get("Name")
+        if not name:
+            continue
+        url_candidates = [metadata.get("Home-page")]
+        for url in metadata.get("Project-URL") or []:
+            url_candidates.append(url.rsplit(",", maxsplit=1)[-1].strip())
+
+        for url in url_candidates:
+            if not url or not url.startswith("https://github.com"):
+                continue
+            _LOGGER.debug("Processing URL: %r", url)
+            url_path_parts = urlparse(url).path.split("/")[1:]
+            if len(url_path_parts) < 2:
+                _LOGGER.warning(
+                    "Skipping URL as GitHub repository and organization cannot be parsed",
+                    url,
+                )
+                continue
+
+            org, repo = url_path_parts[:2]
+            source_url = f"https://github.com/{org}/{repo}"
+
+            response = requests.head(source_url)
+            if response.status_code == 200:
+                git_source_repos[name] = source_url
+            else:
+                _LOGGER.debug("%r is an invalid Github URL", source_url)
+    return git_source_repos  # dictionary
+
+
+def _print_version(ctx: click.Context, _, value: str):
+    """Print adviser version and exit."""
+    if not value or ctx.resilient_parsing:
+        return
+
+    click.echo(__version__)
+    ctx.exit()
+
+
+@click.command()
+@click.pass_context
+@click.option(
+    "-v",
+    "--verbose",
+    is_flag=True,
+    envvar="THOTH_GET_SOURCE_REPOS_DEBUG",
```

Comment:
We use `THOTH_<component>` as a prefix for environment variables. It helps to differentiate to which component the environment variable corresponds to in cluster workflows that use multiple components:

```suggestion
    envvar="THOTH_SOLVER_PROJECT_URL_DEBUG",
```

(here and other options as well)